tasks/message/gasoutputs.go

```diff
@@ -125,6 +132,35 @@ func (p *GasOutputsProcessor) processItem(ctx context.Context, node lens.API, it
 	stop := metrics.Timer(ctx, metrics.ProcessingDuration)
 	defer stop()
 
+	// Note: this item will only be processed if there are receipts for
+	// it, which means there should be a tipset at height+1.  This is only
+	// used to get the destination actor code, so we don't care about side
+	// chains.
```

Comment:
Perhaps its worth adding a check to ensure `child != item`? Alternatively, null rounds can be handled via: https://github.com/filecoin-project/sentinel-visor/pull/214/files#diff-a771f3c826cfe5ebe05c1a003873af47456ee3e16188c1e5d0a1162c0b3e27acR40

Reply:
Given that I just want to extract the dstActorCode, is it a problem if child == parent? The actor code will be the same even if it comes from a slightly wrong state. But maybe I'm missing something?

Reply:
I believe the destination actor may no exist in `child` (when child == parent) if this is the first message being sent to the destination actor. But based on the [below code](https://github.com/filecoin-project/sentinel-visor/pull/270/files#diff-869b68f5505c6de42fcd2d730c0d2e06d684800e092a071b45ff9ff3a385c5adR154-R162) maybe that isn't an issue -- We just won't know the designation actor code if the actor is created in a block preceding a null round.